packages/transaction-controller/src/utils.ts

```diff
@@ -60,38 +62,65 @@ export function validateTxParams(
   txParams: TransactionParams,
   isEIP1559Compatible = true,
 ) {
-  if (
-    !txParams.from ||
-    typeof txParams.from !== 'string' ||
-    !isValidHexAddress(txParams.from)
-  ) {
-    throw rpcErrors.invalidParams(
-      `Invalid "from" address: ${txParams.from} must be a valid string.`,
-    );
-  }
+  validateEIP1559Compatibility(txParams, isEIP1559Compatible);
```

Comment:
I love that this is split into functions, but as we have a few of them, and maybe more in future, is it worth moving to a `validation.ts` for better encapsulation?

Reply:
In a future PR, we could group all these files with a `utils` folder for the functions and a `helpers` folder for the classes like `IncomingTransactionHelper`.

Reply:
Indeed, moved to a `validation.ts`